### ISSUE TITLE
UX around editing the code while in sketch mode.

### ISSUE THREAD

**Issue by Irev-Dev** [KittyCAD/modeling-app#1623]
When your in sketch mode, we do partial executions of just the pipe that makes up the sketch, so it doesn't make sense to let users to edit everything, we might want to introduce the editable sections that we had in the litterbox codeMirror instance.

**Comment by franknoirot** [comment]
I've created a flow for how this could work in [our Sketch Mode Figma page](https://www.figma.com/file/5UUPOawPLEs3D0Gptf42vy/Designs?type=design&node-id=1612-8664&mode=design&t=rhsyq2ppDHdWrjZd-11)

The important step in the flow is here:
![Sketch Flow 01 Step 4](https://github.com/KittyCAD/modeling-app/assets/23481541/c7b3203b-db64-4e93-b696-a7e2b6174b72)

It shows the code pane with an overlay that prevents editing outside of the current sketch, with a little confirm/cancel button group that mirrors the new confirm/cancel button group below the toolbar in the far right. Both of these button groups do the same thing, maintaining our goal of mirrored behavior between code and point-and-click editing. The principle here is that modeling "tools" as we define them–XState nested states, which can be equipped and unequipped—also have the right to isolate parts of the AST for editing.

**Comment by franknoirot** [comment]
This is a feature not a bug, do you mind if I change the label?

Are we still planning on doing this? Are there any alternative solutions to this problem without the need for additional UI, like exiting sketch mode if code outside of it is edited or something?

**Comment by Irev-Dev** [comment]
TLDR keep it as a bug, because I don't think we should build out that UI

https://github.com/user-attachments/assets/8489a251-a906-48ea-a829-310964c435c1

**Comment by Irev-Dev** [comment]
I'm going to close this, because I think it's covered by #6229, in that it handles a number of "manual edit while in sketch mode situations" and exits out of sketch mode with a toast of situations it can't handle.

lmk if you disagree.
